api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -50,52 +51,56 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	 */
 	@Override
 	public Cohort getCohort(Integer id) throws DAOException {
-		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
+		return sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
 
 	/**
 	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(Integer, boolean, Date)
 	 */
 	@Override
-	@SuppressWarnings("unchecked")
 	public List<Cohort> getCohortsContainingPatientId(Integer patientId, boolean includeVoided,
-	                                                  Date asOfDate) throws DAOException {
-		Disjunction orEndDate = Restrictions.disjunction();
-		orEndDate.add(Restrictions.isNull("m.endDate"));
-		orEndDate.add(Restrictions.gt("m.endDate", asOfDate));
+													  Date asOfDate) throws DAOException {
+		Session session = sessionFactory.getCurrentSession();
+		CriteriaBuilder cb = session.getCriteriaBuilder();
+		CriteriaQuery<Cohort> query = cb.createQuery(Cohort.class);
```

Comment:
For the sake of consistency, between CriteriaBuilder and CriteriaQuery, why not any of these?
1) cb and cq
2) builder and query
3) criteriaBuilder and criteriaQuery

Reply:
Updated to use option 1 (cb and cq)